gmso/formats/lammpsdata.py

```diff
@@ -199,3 +209,228 @@ def write_lammpsdata(topology, filename, atom_style='full'):
         # TODO: Write out bonds
         # TODO: Write out angles
         # TODO: Write out dihedrals
+
+
+def read_lammpsdata(filename, atom_style='full', unit_style='real', potential='lj'):
+    """
+    Read in a lammps data file as a GMSO topology
+
+    Parameters
+    ----------
+    filename : str
+        LAMMPS data file
+    atom_style : str, optional, default='full'
+        Inferred atom style defined by LAMMPS
+    potential: str, optional, default='lj'
+        Potential type defined in data file
+
+    Returns
+    -------
+    top : GMSO Topology
+        A GMSO Topology object
+
+    Notes
+    -----
+    See http://lammps.sandia.gov/doc/2001/data_format.html for a full description of the LAMMPS data format.  
+    Currently only supporting LJ potential parameters.
+    """
+    # TODO: Add argument to ask if user wants to infer bond type
+    top = Topology()
+
+    # Validate 'atom_style'
+    if atom_style not in ['full']:
```

Comment:
this works now, but something that might be a bit easier to handle later is a list of acceptable `atom_styles` we check against, similar to the way we handle the potential templates for various writers.